### PR TITLE
New Plugin Viper Search IOC in Viper Database

### PR DESCRIPTION
**Pull request by IrootGeek**
The plugin Viper can Search IOC [ MD5, SHA256 ] in Viper Database

### CODE REVIEW

File: modules/viper.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>.
+
+import requests
+
+from lib.io import module as mod
+
+
+class viper:
```
Comment (rmarsollier) [inline modules/viper.py:26]: class name needs upercase first letter

File: modules/viper.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        """
+        Viper API Connection
+        """
+        server = self.config["viper_server"]+"file/find"
+        if self.type == "MD5":
+            ioc = "md5=%s" % self.ioc
+        if self.type == "SHA256":
+            ioc = "sha256=%s" % self.ioc
+
+        respond = requests.post(server,headers={'User-agent': 'viper Connector'}, data= ioc)
```
Comment (rmarsollier) [inline modules/viper.py:53]: Use user-agent defined in config.ini

File: modules/viper.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+from lib.io import module as mod
+
+
+class viper:
+    def __init__(self, ioc, type, config):
+        self.config = config
+        self.module_name = __name__.split(".")[1]
+        self.types = ["MD5", "SHA256"]
+        self.search_method = "Online"
```
Comment (rmarsollier) [inline modules/viper.py:31]: switch to offline, viper servers are expected to be on local networks

**Review by rmarsollier — CHANGES_REQUESTED** [review]
Some minor things to change.

**Comment by IrootGeek** [comment]
I update viper plugin

**Pull request closed without merging**